python/lsst/meas/extensions/scarlet/scarletDeblendTask.py

```diff
@@ -1196,7 +1321,81 @@ def _skipParent(self, parent, skipKey, logMessage):
         parent.set(self.deblendSkippedKey, True)
         parent.set(skipKey, True)
 
-    def _updateParentRecord(self, parent, nPeaks, nChild,
+    def _isSkipped(self, parent, mExposure):
```

Comment:
This is just git being weird. The `_updateParentRecord` method is still there and was refactored to somewhere else, but for some reason git did the diff this way.

Reply:
I think `_check` also doesn't imply any updates. I see that you use this function in both `setSkipFlags`, where it's only called for the side effect without checking the return value, and elsewhere only to check the return value. So if you want to keep it one function, it should reflect the fact that it can change `parent` if necessary. Otherwise, I suppose you could have it do something like return either `None` or `(skipKey, skipMessage)` and not do the modifications itself.

Reply:
I used the second option and kept it in its own commit. This gave me my first opportunity to use the walrus operator, but I'll wait for your approval before re-running Jenkins and merging.

Reply:
That seems like it should work. I generally prefer the explicit `(skipArgs := self._checkSkipped(src, mExposure)) is None` to the version without `is None` (you have one instance of each).

Reply:
Yep, that was me being careless. Thanks for catching it.